### PR TITLE
Bugfix/fetch images from icloud

### PR DESCRIPTION
**Pull request by danfordham**

### CODE REVIEW

File: Sources/Model/PhotoCollectionViewDataSource.swift [hunk 1]
```diff
@@ -63,18 +63,23 @@ final class PhotoCollectionViewDataSource : NSObject, UICollectionViewDataSource
         if let settings = settings {
             cell.settings = settings
         }
-        
+
+        // Clear any previously shown image
+        cell.imageView.image = nil
```
Comment (danfordham) [inline Sources/Model/PhotoCollectionViewDataSource.swift]: This prevents the previous image being seen if a request is delayed, which can happen when scrolling quickly through a large album, that is fetching images from the cloud.

File: Sources/Model/PhotoCollectionViewDataSource.swift [hunk 2]
```diff
@@ -63,18 +63,23 @@ final class PhotoCollectionViewDataSource : NSObject, UICollectionViewDataSource
...
         if cell.tag != 0 {
-            photosManager.cancelImageRequest(PHImageRequestID(cell.tag))
+            photosManager.cancelImageRequest(PHImageRequestID(Int32(cell.tag)))
         }
         
         let asset = fetchResult[indexPath.row]
         cell.asset = asset
         
         // Request image
         cell.tag = Int(photosManager.requestImage(for: asset, targetSize: imageSize, contentMode: imageContentMode, options: imageRequestOptions) { (result, _) in
```
Comment (danfordham) [inline Sources/Model/PhotoCollectionViewDataSource.swift]: This is called when a request is cancelled, and the image was being set to nil, looks like this was causing some cells to be 'blank', as well as the nil image request options.

File: BSImagePicker.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -41,7 +41,6 @@
 		55BCF8D521D52C1000386752 /* PhotoCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 55BCF8C121D52C1000386752 /* PhotoCell.swift */; };
 		55BCF8D621D52C1000386752 /* AlbumCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 55BCF8C221D52C1000386752 /* AlbumCell.swift */; };
 		55BCF8D721D52C1000386752 /* SelectionView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 55BCF8C321D52C1000386752 /* SelectionView.swift */; };
```
Comment (danfordham) [inline BSImagePicker.xcodeproj/project.pbxproj]: Decided to delete AlbumTitleView, as after the title sizing fix most of the code could be set during initialisation, and isn't needed when setting the title.

File: Sources/Model/AlbumTableViewDataSource.swift [hunk 4]
```diff
@@ -56,7 +56,12 @@ final class AlbumTableViewDataSource : NSObject, UITableViewDataSource {
         let album = fetchResults[indexPath.section][indexPath.row]
         // Title
         cell.albumTitleLabel.text = album.localizedTitle
-        
+
```
Comment (danfordham) [inline Sources/Model/AlbumTableViewDataSource.swift]: Didn't see any issues with the album table, but decided to add this to bring it into line with the change in PhotoCollectionViewDataSource. If a cell is re-used, and the image request is delayed, the previous/wrong image won't be shown.

Comment (mikaoj) [reply Sources/Model/AlbumTableViewDataSource.swift]: This is already being done in prepareForReuse on AlbumCell.

File: Sources/Model/PhotoCollectionViewDataSource.swift (hunk 1 above)
Comment (mikaoj) [reply Sources/Model/PhotoCollectionViewDataSource.swift]: I'll move this to PhotoCells prepareForReuse

File: Sources/Model/PhotoCollectionViewDataSource.swift [hunk 5]
```diff
@@ -63,18 +63,23 @@ final class PhotoCollectionViewDataSource : NSObject, UICollectionViewDataSource
...
+          if let result = result {
```
Comment (mikaoj) [inline Sources/Model/PhotoCollectionViewDataSource.swift]: Yeah, I didn't realise the closure was called even on cancelation. Nice! :)

**Pull request merged**